src/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableArray_1.cs

```diff
@@ -790,6 +793,12 @@ public ImmutableArray<T> RemoveAt(int index)
         [Pure]
         public ImmutableArray<T> RemoveRange(int index, int length)
         {
+            // TODO: There is an inconsistency here between what type of exception will be thrown
+            // if we are an IsDefault instance.
+            // If the index is less than 0, the && below will short-circuit, and self.Length will
+            // not be evaluated. Then we will get an ArgumentOutOfRangeException.
+            // Otherwise, we will get a NullReferenceException when trying to evaluate self.Length.
+
             var self = this;
             Requires.Range(index >= 0 && index <= self.Length, nameof(index));
```

Comment:
Either exception should never be caught and is a sign of an error in the caller. So consistently prioritizing one over the other sounds like (mild) goodness to me.
But unless it's blocking your test development, I suggest you fix this in a separate PR as well.